browser-test/src/support/session_timeout.ts

```diff
@@ -0,0 +1,228 @@
+
+    // Create a mock date that will be ahead by the specified milliseconds
+    const mockTimeOffset = ms
+
+    const realNow = originalNow.call(Date)
+    const mockedNow = realNow + mockTimeOffset
+
+    console.log(
+      `Mocking time: Real time: ${realNow / 1000}, Mocked time: ${mockedNow / 1000}, Offset: ${mockTimeOffset / 1000}`,
+    )
+
+    // Override the Date constructor
+    // @ts-expect-error - Intentionally mocking Date constructor
+    window.Date = class extends OriginalDate {
+      constructor(...args: ConstructorParameters<typeof OriginalDate>) {
+        if (args.length > 0) {
+          // If arguments provided, behave normally
+          super(...args)
+        } else {
+          // When called with no arguments, return current time + offset
+          super(realNow + mockTimeOffset)
+        }
+      }
+
+      // Override getTime to return the advanced time
+      getTime() {
+        return realNow + mockTimeOffset
+      }
+
+      // Override now to return the advanced time
+      static now() {
+        return realNow + mockTimeOffset
+      }
+    }
+
+    // Dispatch timechange event with time details. Simply advancing time does not trigger the timeout
+    // that is already set up in the browser
+    const timeChangeEvent = new CustomEvent('timechange', {
+      detail: {
+        realTime: realNow,
+        mockedTime: mockedNow,
+        offset: mockTimeOffset,
+      },
+    })
+
+    console.log(
+      'Dispatching timechange event with details:',
+      timeChangeEvent.detail,
+    )
+    window.dispatchEvent(timeChangeEvent)
+  }, advanceMilliseconds)
+  // Give the browser a moment to process the event
+  await page.waitForTimeout(100)
+}
+
+/**
+ * Advances time to trigger a specific timeout event
+ */
+async function advanceToTimeout(page: Page, timeoutType: TimeoutType) {
+  // Get the adjusted timeout data that accounts for clock skew
+  const data = await getAdjustedTimeoutData(page)
+  if (!data) {
+    throw new Error('No timeout data found')
+  }
+
+  // Get current browser time
+  const browserNow = await page.evaluate(() => Math.floor(Date.now() / 1000))
+
+  // Get target time based on timeout type
+  const targetTime = data[timeoutType]
+  console.log(`timeoutType: ${timeoutType}, timeoutValue: ${targetTime}`)
+  // Add small buffer (2 seconds) to ensure we're past the threshold
+  const advanceMs = (targetTime - browserNow + 2) * 1000
+
+  console.log(`Advancing time by ${advanceMs}ms to trigger ${timeoutType}`)
+  await mockDateAndAdvanceTime(page, advanceMs)
+}
+
+export async function advanceToSessionInactivityWarning(page: Page) {
+  return advanceToTimeout(page, 'inactivityWarning')
+}
+
+export async function advanceToSessionLengthWarning(page: Page) {
+  return advanceToTimeout(page, 'totalWarning')
+}
+
+export async function advanceToSessionTimeout(page: Page) {
+  return advanceToTimeout(page, 'totalTimeout')
+}
+
+export async function setupSessionTimeout({
```

Comment:
```suggestion

```
(blocking) This function can be removed. Please use `await enableFeatureFlag(page, 'session_timeout_enabled')` to set the feature flags.